src/app/components/ChooseSeats.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC } from 'react';
+import { Seat } from '../(site)/tickets/page';
+
+type Props = {
+  seatsArray: [] | { status: Seat; seatNumber: number }[];
+  ticketCount: number;
+  onSetSeatsArray: (
+    newArray: {
+      status: Seat;
+      seatNumber: number;
+    }[]
+  ) => void;
+};
+
+const ChooseSeats: FC<Props> = ({
```

Comment:
This is a complicated component, but that's ok, because it's separated and you don't have to worry about it's internals to use it.